wheatley/bot.py

```diff
@@ -137,7 +140,17 @@ def _on_row_gen_change(self, row_gen_json: JSON) -> None:
 
     def _on_size_change(self) -> None:
         self._check_number_of_bells()
+        self._opening_rounds = generate_starting_row(
+            self.number_of_bells, self.row_generator.custom_start_row)
         self._rounds = rounds(self.number_of_bells)
+        self._check_starting_row()
+
+    def _check_starting_row(self) -> bool:
+        if len(self._opening_rounds) != self.number_of_bells:
+            self.logger.warning(f"Row generation requires {self.row_generator.stage} "
+                                + "bells in the starting row Wheatley will not ring!")
+            return False
+        return True
```

Comment:
I think the error message should refer to `self._opening_rounds` as that's the given part. Something like
```suggestion
    def _check_starting_row(self) -> bool:
        if len(self._opening_rounds) > self.number_of_bells:
            self.logger.warning(f"The current tower has fewer bells ({self._tower.number_of_bells})"
            + "than the starting row {self._opening_rounds}. Wheatley will not ring!")
            return False
        return True
```

Using `len(self._opening_rounds) > self.number_of_bells` as the not ringing condition would allow `13524` as a starting row for Plain Bob Doubles, but it doesn't save much over typing the cover bell.